Changelog.md

```diff
@@ -23,6 +24,8 @@ Bugfixes:
 
 
 AST Changes:
+ * AST: Add ``isSuffixCall`` field to ``FunctionCall`` AST node, which is ``true`` for calls resulting from applying a literal suffix. Additionally, the ``arguments`` annotation is omitted from such notes.
```

Comment:
The latter is not a visible AST change, is it? I.e.
```suggestion
 * AST: Add ``isSuffixCall`` field to ``FunctionCall`` AST node, which is ``true`` for calls resulting from applying a literal suffix.
```

Reply:
Let me check.

You reminded me that I actually wanted to check if by default it's `null` or an empty list (the former would make it not show up by default). If it's not in the AST at all then that's moot too I guess.

Reply:
No, `arguments` is included in the AST for `FunctionCall` nodes: https://github.com/ethereum/solidity/blob/4a8d6618f5b398077f694835905f93d0a3890289/libsolidity/ast/ASTJsonExporter.cpp#L884

Unless you mean that despite this we don't consider it an AST change?

Also, weirdly, I just noticed that my ASTJSON test with a suffix call does have that field in its `FunctionCall` node. Need to check that. This should not be happening unless I'm misunderstanding something here.

Reply:
> I.e. note that this is not `make_pair("arguments", toJson(_node.annotation().arguments)),`

Yeah, just figured that out after stepping though the code in the debugger. It was in my blind spot because I didn't expect there to be two different things called "arguments". Turns out is that we have both an annotation called `arguments` and `FunctionCall::arguments()` (which is not a getter for the annotation). 

Especially the fact the annotation is not exported is a total surprise to me, because I assumed that we pretty much always export annotations (at some point I even used to think it happens automatically).

There are so many slightly different things that all could be referred to as "arguments" in our code that we'd really be better off having more precise names for them. E.g. `ExpressionAnnotation::argumentNamesAndTypes` and `FunctionCall::argumentExpressions()`.

In any case, you're right about the changelog. I removed that bit. Thanks for noticing it.

Reply:
We definitely don't generally export annotations - rather the opposite, the base AST is what we *always* export. Annotations we only export if they provide valuable additional information for tooling - while the expression arguments annotation (and a lot of the annotations in general) are just an internal convenience store of information that is already otherwise present in the AST in all cases.